# Simulation settings for using moveit_sim_controllers
moveit_sim_hw_interface:
  joint_model_group: todo_group_name
  joint_model_group_pose: todo_state_name
# Settings for ros_control_boilerplate control loop
generic_hw_control_loop:
  loop_hz: 300
  cycle_time_error_threshold: 0.01
# Settings for ros_control hardware interface
hardware_interface:
  joints:
    - $(arg tf_prefix)shoulder_pan_joint
    - $(arg tf_prefix)shoulder_lift_joint
    - $(arg tf_prefix)elbow_joint
    - $(arg tf_prefix)wrist_1_joint
    - $(arg tf_prefix)wrist_2_joint
    - $(arg tf_prefix)wrist_3_joint
  sim_control_mode: 1  # 0: position, 1: velocity
# Publish all joint states
# Creates the /joint_states topic necessary in ROS
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50
arm_controller:
  type: position_controllers/JointTrajectoryController
  joints:
    - $(arg tf_prefix)shoulder_pan_joint
    - $(arg tf_prefix)shoulder_lift_joint
    - $(arg tf_prefix)elbow_joint
    - $(arg tf_prefix)wrist_1_joint
    - $(arg tf_prefix)wrist_2_joint
    - $(arg tf_prefix)wrist_3_joint
  pid_gains:
    $(arg tf_prefix)shoulder_pan_joint: {p: 100.0, i: 0.01, d: 10.0}
    $(arg tf_prefix)shoulder_lift_joint: {p: 100.0, i: 0.01, d: 10.0}
    $(arg tf_prefix)elbow_joint: {p: 100.0, i: 0.01, d: 10.0}
    $(arg tf_prefix)wrist_1_joint: {p: 100.0, i: 0.01, d: 10.0}
    $(arg tf_prefix)wrist_2_joint: {p: 100.0, i: 0.01, d: 10.0}
    $(arg tf_prefix)wrist_3_joint: {p: 100.0, i: 0.01, d: 10.0}
    $(arg tf_prefix)left_wheel_joint: {p: 100.0, i: 0.01, d: 10.0}
    $(arg tf_prefix)right_wheel_joint: {p: 100.0, i: 0.01, d: 10.0}
  constraints:
    goal_time: 0.6
    stopped_velocity_tolerance: 0.05
    $(arg tf_prefix)shoulder_pan_joint: {trajectory: 0.1, goal: 0.1}
    $(arg tf_prefix)shoulder_lift_joint: {trajectory: 0.1, goal: 0.1}
    $(arg tf_prefix)elbow_joint: {trajectory: 0.1, goal: 0.1}
    $(arg tf_prefix)wrist_1_joint: {trajectory: 0.1, goal: 0.1}
    $(arg tf_prefix)wrist_2_joint: {trajectory: 0.1, goal: 0.1}
    $(arg tf_prefix)wrist_3_joint: {trajectory: 0.1, goal: 0.1}
  stop_trajectory_duration: 0.5
  state_publish_rate:  25
  action_monitor_rate: 10
# controller_list:
#   - name: arm_controller
#     action_ns: follow_joint_trajectory
#     default: True
#     type: FollowJointTrajectory
#     joints:
#       - $(arg tf_prefix)shoulder_pan_joint
#       - $(arg tf_prefix)shoulder_lift_joint
#       - $(arg tf_prefix)elbow_joint
#       - $(arg tf_prefix)wrist_1_joint
#       - $(arg tf_prefix)wrist_2_joint
#       - $(arg tf_prefix)wrist_3_joint
# gazebo_ros_control:
#   pid_gains:
#     $(arg tf_prefix)shoulder_pan_joint: {p: 100.0, i: 0.01, d: 10.0}
#     $(arg tf_prefix)shoulder_lift_joint: {p: 100.0, i: 0.01, d: 10.0}
#     $(arg tf_prefix)elbow_joint: {p: 100.0, i: 0.01, d: 10.0}
#     $(arg tf_prefix)wrist_1_joint: {p: 100.0, i: 0.01, d: 10.0}
#     $(arg tf_prefix)wrist_2_joint: {p: 100.0, i: 0.01, d: 10.0}
#     $(arg tf_prefix)wrist_3_joint: {p: 100.0, i: 0.01, d: 10.0}
#     $(arg tf_prefix)left_wheel_joint: {p: 100.0, i: 0.01, d: 10.0}
#     $(arg tf_prefix)right_wheel_joint: {p: 100.0, i: 0.01, d: 10.0}  